Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -848,9 +848,12 @@ void RenderElement::styleWillChange(StyleDifference diff, const RenderStyle& new
         bool willBeVisible = newStyle.visibility() == Visibility::Visible && !newStyle.skippedContentReason().has_value();
         if (wasVisible != willBeVisible) {
             if (CheckedPtr layer = enclosingLayer()) {
-                if (willBeVisible)
-                    layer->setHasVisibleContent();
-                else if (layer->hasVisibleContent() && (this == &layer->renderer() || layer->renderer().style().visibility() != Visibility::Visible))
+                if (willBeVisible) {
+                    if (m_style.visibility() != newStyle.visibility())
+                        layer->setHasVisibleContent();
```

Comment:
Does this do the right thing if both visibility, and the skipped-ness changed?

Reply:
Ok, my modifications yesterday were wrong, it is possible to change visibility hidden to visible as well as unskipping skipped content in the same style update using the test below, I tried it and it passes (green rect on top is visible), since in this case setHasVisibleContent will not do an early return (m_hasVisibleContent is false).

```
<!doctype HTML>
<html class="reftest-wait">
<meta charset="utf8">
<title>Content Visibility: auto, scroll away and back</title>
<link rel="author" title="Rob Buis" href="mailto:rbuis@igalia.com">
<link rel="help" href="https://drafts.csswg.org/css-contain/#content-visibility">
<link rel="match" href="content-visibility-096-ref.html">
<meta name="assert" content="scolling away from c-v: auto container and back renders as expected">
<script src="/common/reftest-wait.js"></script>

<style>
.spacer {
  height: 10000px;
}
#child {
  width: 100px;
  height: 100px;
  background: green;
  position: relative
}
#target {
  width: 10px;
  height: 10px;
}
</style>

<div id="container" style="content-visibility: auto">
  <div id=child></div>
</div>
<div class=spacer></div>
<div id=target></div>

<script>

function step1() {
  document.getElementById("target").scrollIntoView(true /* alignToTop */);
  requestAnimationFrame(step2);
}

function step2() {
  container.style.visibility = "hidden";
  requestAnimationFrame(finishTest);
}

function finishTest() {
  container.setAttribute("style", "content-visibility: visible; visibility: visible");
  window.scrollTo(0, 0);
  requestAnimationFrame(takeScreenshot);
}

window.onload = requestAnimationFrame(() => {
  requestAnimationFrame(() => {
    requestAnimationFrame(() => {
      step1();
    });
  });
});

</script>
</html>
```

Reply:
I now included above test as wpt test for completeness.